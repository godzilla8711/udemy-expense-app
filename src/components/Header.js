import React from 'react';
import { NavLink } from 'react-router-dom';

class Header extends React.Component {
  render() {
    return (
      <div>
        <h2>Expense App</h2>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add Evidence</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit Evidence</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </div>
    );
  }
}

export default Header;
