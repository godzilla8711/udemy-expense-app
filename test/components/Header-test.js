import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';

import Header from '../../src/components/Header';

describe('render', () => {
  it('renders Header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('NavLink').length).to.equal(4);
    // expect(wrapper).toMatchSnapshot();
  });
});
