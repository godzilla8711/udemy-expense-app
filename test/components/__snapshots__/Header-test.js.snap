// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders component 1`] = `
<div>
  <h2>
    Expense App
  </h2>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Add Evidence
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Edit Evidence
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
